src/SideNav.js

```diff
@@ -1,21 +1,81 @@
-import React, { PropTypes } from 'react';
-import Icon from './Icon';
+import React, { Component, PropTypes } from 'react';
 import idgen from './idgen';
 
-const SideNav = ({ children, id = [`sidenav_${idgen()}`] }) =>
-  <nav>
-    <ul className='right hide-on-med-and-down'>{children}</ul>
-    <ul id={id} className='side-nav'>{children}</ul>
-    <a href='#' data-activates={id} className='button-collapse'>
-      <Icon>view_headline</Icon>
-    </a>
-  </nav>;
+class SideNav extends Component {
+  componentDidMount () {
+    const { options = {}, right, closeOnClick } = this.props;
+    $(this._trigger).sideNav({
+      ...options,
+      edge: right || options.edge,
+      closeOnClick: closeOnClick || options.closeOnClick
+    });
+  }
+
+  componentWillUnmount () {
+    $(this._trigger).sideNav('hide');
+  }
+
+  render () {
+    const { children, id = [`sidenav_${idgen()}`], ...props } = this.props;
+    this.id = id;
+    delete props.right;
+    delete props.closeOnClick;
+    delete props.trigger;
+    delete props.options;
+    return (
+      <span>
+        { this.renderTrigger() }
+        <ul id={id} className='side-nav' {...props}>
+          {children}
+        </ul>
+      </span>
+    );
+  }
+
+  renderTrigger () {
+    const { trigger } = this.props;
+    return React.cloneElement(trigger, {
+      ref: (t) => (this._trigger = `[data-activates=${this.id}]`),
+      'data-activates': this.id
+    });
+  }
+}
 
 SideNav.propTypes = {
-  children: PropTypes.node,
+  /**
+   * sidenav id. If none is passed, an id will be generated.
+   */
   id: PropTypes.string,
+  /**
+   * Component that is rendered to trigger the sidenav
+   */
+  trigger: PropTypes.node.isRequired,
+  /**
+   * Position the sidenav on the right edge of the screen
+   * @default false
+   */
   right: PropTypes.bool,
-  left: PropTypes.bool
+  /**
+   * Closes side-nav on anchor clicks
+   * @default false
+   */
+  closeOnClick: PropTypes.bool,
+  /**
+   * Options hash for the sidenav.
+   * More info: http://materializecss.com/side-nav.html#options
+   */
+  options: PropTypes.shape({
+    menuWidth: PropTypes.number,
+    edge: PropTypes.oneOf(['left', 'right']),
+    closeOnClick: PropTypes.bool,
+    draggable: PropTypes.bool
+  }),
+  children: PropTypes.node
+};
+
+SideNav.defaultProps = {
+  right: false,
+  closeOnClick: false
```

Comment:
I meant: the default values show up in the documentation only if you add defaultProps. If thats not needed then this can be removed.